include/networkit/auxiliary/GlobalState.hpp

```diff
@@ -0,0 +1,71 @@
+#ifndef NETWORKIT_AUXILIARY_GLOBAL_STATE_HPP_
+#define NETWORKIT_AUXILIARY_GLOBAL_STATE_HPP_
+
+// networkit-format
+
+#include <atomic>
+#include <csignal>
+#include <cstdint>
+#include <fstream>
+#include <mutex>
+
+#include <networkit/auxiliary/Log.hpp>
+#include <networkit/auxiliary/SignalHandling.hpp>
+
+#include <networkit/networkit_export.h>
+
+namespace Aux {
+
+namespace GlobalState {
+
+namespace Random {
+
+NETWORKIT_EXPORT uint64_t getSeed();
+NETWORKIT_EXPORT void setSeed(uint64_t seed);
+
+NETWORKIT_EXPORT uint64_t getGlobalSeed();
+NETWORKIT_EXPORT void incGlobalSeed();
+
+NETWORKIT_EXPORT void setSeedUseThreadId(bool useThreadId);
+NETWORKIT_EXPORT bool getSeedUseThreadId();
```

Comment:
I wonder if this is a good time to get rid of non-global seeds. Are there any users of `setSeed()` without also specifying `useThreadId` as `true`?

Reply:
Yeah, we should do it. However it is better to move it to another PR.